include/microhttpd_retro.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 Emmanuel Engelhart <kelson@kiwix.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU  General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+/**
+ * @author Christian Grothoff
+ *
+ * @file
+ * MHD compatibility definitions for warning-less compile of
+ * our code against MHD before and after #MHD_VERSION 0x00097002.
+ */
+#include <microhttpd.h>
+
+#if MHD_VERSION >= 0x00097002
+/**
+ * Data type to use for functions return an "MHD result".
+ */
+#define MHD_RESULT enum MHD_Result
+
+#else
+
+/**
+ * Data type to use for functions return an "MHD result".
+ */
+#define MHD_RESULT int
+
+#endif
```

Comment:
I would have simply does :
```
#if MHD_VERSION < 0x00097002
#define MHD_Result int
#endif
```

This way we simply would have to remove the 3 lines once we don't want compatibility with old libmicrohttpd. And we don't introduce a new identifier.
But I would not block for that.

Reply:
Bien meilleur en effet.